tests/unit/tests_high_five_base.cpp

```diff
@@ -426,6 +426,56 @@ TEST_CASE("Test groups and datasets") {
     }
 }
 
+#if H5_VERSION_GE(1, 10, 1)
+TEST_CASE("FileSizeAndUnusedSpace") {
+    const std::string file_name_untracked("filesize_untracked_unused.h5");
+    const std::string file_name_tracked("filesize_tracked_unused.h5");
+    const std::string ds_path("/dataset");
+    const std::vector<int> data{13, 24, 36};
+    const size_t untracked_file_size = 2048;
+    const size_t untracked_unused_space = 600;
+    const size_t tracked_file_size = 1158;
+    const size_t tracked_unused_space = 103;
+
+
+    {
+        File file(file_name_untracked, File::ReadWrite | File::Create | File::Truncate);
```

Comment:
FYI this can be reduced down to:
```
File file(file_name_untracked, File::Truncate);
```
It's consistent with the rest of the file so there's no reason to change it.